BlazarService/src/main/java/com/hubspot/blazar/resources/InstantMessageResource.java

```diff
@@ -0,0 +1,88 @@
+package com.hubspot.blazar.resources;
+
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import com.hubspot.blazar.externalservice.slack.SlackChannel;
+import com.hubspot.blazar.base.notifications.InstantMessageConfiguration;
+import com.hubspot.blazar.data.service.InstantMessageConfigurationService;
+import com.hubspot.blazar.integration.slack.SlackClient;
+import com.hubspot.jackson.jaxrs.PropertyFiltering;
+
+@Path("/instant-message")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class InstantMessageResource {
+
+
+  private final SlackClient slackClient;
+  private InstantMessageConfigurationService instantMessageConfigurationService;
+
+  @Inject
+  public InstantMessageResource(SlackClient slackClient,
+                                InstantMessageConfigurationService instantMessageConfigurationService) {
+    this.slackClient = slackClient;
+    this.instantMessageConfigurationService = instantMessageConfigurationService;
+  }
+
+  @GET
+  @PropertyFiltering
+  @Path("/configurations")
+  public Set<InstantMessageConfiguration> getAll() {
+    return instantMessageConfigurationService.getAll();
+  }
+
+  @PUT
+  @Path("/configurations")
+  public Optional<InstantMessageConfiguration> insert(InstantMessageConfiguration instantMessageConfiguration) {
+    long id = instantMessageConfigurationService.insert(instantMessageConfiguration);
+    return instantMessageConfigurationService.get(id);
+  }
+
+  @POST
+  @Path("/configurations")
```

Comment:
Conventionally you POST to insert a new resource object and you PUT (is idempotent) to an ID to update an existing resource with that Id. So when you PUT you better go for: 
PUT /configurations/id (more clean that you update an existing resource)
You can also PUT /configurations as you do above but you need to check if the posted object has an id already and throw an exception if it hasn't, I would prefer the former approach.